src/tools/compiletest/src/runtest.rs

```diff
@@ -3239,8 +3239,19 @@ impl<'test> TestCx<'test> {
     }
 
     fn diff_mir_files(&self, before: PathBuf, after: PathBuf) -> String {
-        let before = self.get_mir_dump_dir().join(before);
-        let after = self.get_mir_dump_dir().join(after);
+        let to_full_path = |path: PathBuf| {
+            let full = self.get_mir_dump_dir().join(&path);
+            if !full.exists() {
+                panic!(
```

Comment:
👍 there are a couple other places in `check_mir_dump` that also produce confusing / less useful errors but that could be fixed in a seperate pr.